src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml.cs

```diff
@@ -258,6 +277,36 @@ protected override void OnKeyUp(KeyEventArgs e)
             }
         }
 
+        private void SetComboboxCurrentVersion(string comboboxText, bool userTypedAVersionRange, IEnumerable<NuGetVersion> versions)
+        {
+            NuGetVersion matchVersion = null;
+            VersionRange userRange = null;
+
+            // Get the best version from the range if the user typed a custom version
+            bool userTypedAValidVersionRange = VersionRange.TryParse(comboboxText, out userRange);
+            if (userTypedAValidVersionRange && (userTypedAVersionRange || (userRange != null && userRange.IsFloating)))
```

Comment:
I'm not sure I follow this particular logic:
`(userTypedAValidVersionRange && (userTypedAVersionRange`
Why would it matter if it's a valid `VersionRange`, but not a range containing () or [] ?

Reply:
The problem here is that when you try to parse for example "5", it will be returned as a valid range, I was thinking that was not a desired behavior, now that I think about it if the parser lets us do that then it should be correct